src/test/shared/settingsConfiguration.test.ts

```diff
@@ -328,4 +328,25 @@ describe('Experiments', function () {
         await sut.update('jsonResourceModification', true)
         assert.strictEqual(await sut.isExperimentEnabled('jsonResourceModification'), true)
     })
+
+    it('fires events from nested settings', async function () {
+        const info = vscode.workspace.getConfiguration().inspect('aws.experiments.jsonResourceModification')
```

Comment:
Problem with using the real settings API is that the developer's settings can affect behavior :(

But that's fine. CI will always have a clean slate.

Reply:
> Problem with using the real settings API is that the developer's settings can affect behavior :(

There are ways around that. Either vscode should allow us to specify a settings file, or we could save/restore the local settings.json.

I'd prefer to wait (forever) for vscode to fix their model though. Meanwhile CI is fine, yes.

Reply:
> Either vscode should allow us to specify a settings file

In this case I'm specifying the workspace settings, but the way the settings API works is that `get` pulls in settings from _everywhere_ unless a scope is specified. Maybe if I use a workspace folder scope it'll work.

Reply:
So even if you specify a scope it still merges things for `get`. This may be a VSC bug because I would expect that I only get values set in the workspace `settings.json` if I provide a scope.

Reply:
yep, it's tricky and maybe even buggy: https://github.com/aws/aws-toolkit-vscode/commit/6884b78579830db5a450715f0a43c3e224ef6e24